src/coreclr/jit/morph.cpp

```diff
@@ -16922,6 +16922,34 @@ void Compiler::fgMorphStmts(BasicBlock* block, bool* lnot, bool* loadw)
 
     fgCurrentlyInUseArgTemps = hashBv::Create(this);
 
+    // Check if the current block is a candidate to be converted to BBJ_THROW
+    if (block->bbJumpKind != BBJ_THROW)
```

Comment:
You've introduced a full IR walk just to set this flag. Is it worth it? What's the throughput impact? Can it be done with any existing IR walks?

You've also introduced a new BBF flag. We have space, yes, but every new flag introduces some new complexity. If it's required, then ok.